src/components/ProjectCard/ProjectCardBasicsForm.tsx

```diff
@@ -2,46 +2,43 @@ import useProjectCardBasicsForm from '@/hooks/useProjectCardBasicsForm';
 import { useAppDispatch, useAppSelector } from '@/hooks/common';
 import { IAppForms, IProjectCardBasicsForm } from '@/interfaces/formInterfaces';
 import { FC, memo, useCallback } from 'react';
-import { SubmitHandler } from 'react-hook-form';
 import { FormFieldCreator } from '../shared';
-import { Button } from 'hds-react/components/Button';
-import { patchProjectCardThunk } from '@/reducers/projectCardSlice';
+import { silentPatchProjectCardThunk } from '@/reducers/projectCardSlice';
 import { IProjectCardRequest } from '@/interfaces/projectCardInterfaces';
-import './basicsFormStyles.css';
 import { RootState } from '@/store';
-import { emptyStringsToNull, getOptionId } from '@/utils/common';
+import { dirtyFieldsToRequestObject } from '@/utils/common';
+import './basicsFormStyles.css';
+import { FieldValues, SubmitHandler } from 'react-hook-form';
 
 const ProjectCardBasicsForm: FC = () => {
   const dispatch = useAppDispatch();
-  const { handleSubmit, formFields } = useProjectCardBasicsForm();
+  const { formFields, formMethods } = useProjectCardBasicsForm();
   const projectId = useAppSelector((state: RootState) => state.projectCard.selectedProjectCard)?.id;
 
-  const onSubmit: SubmitHandler<IProjectCardBasicsForm> = useCallback(
-    async (form: IProjectCardBasicsForm) => {
-      const { type, area, ...formData } = emptyStringsToNull(
-        form as IAppForms,
-      ) as IProjectCardBasicsForm;
-
-      const data: IProjectCardRequest = {
-        ...formData,
-        type: getOptionId(type),
-        area: getOptionId(area),
-      };
+  const {
+    formState: { dirtyFields, isDirty },
+    handleSubmit,
+  } = formMethods;
 
-      if (projectId) {
-        dispatch(patchProjectCardThunk({ id: projectId, data }));
+  const onSubmit = useCallback(
+    (form: IProjectCardBasicsForm) => {
+      if (isDirty) {
+        const data: IProjectCardRequest = dirtyFieldsToRequestObject(
+          dirtyFields,
+          form as IAppForms,
+        );
+        projectId && dispatch(silentPatchProjectCardThunk({ id: projectId, data }));
       }
     },
-    [dispatch, projectId],
+    [dirtyFields, projectId, dispatch, isDirty],
   );
 
   return (
     <div className="basics-form">
-      <form onSubmit={handleSubmit(onSubmit)}>
+      <form onBlur={handleSubmit(onSubmit) as SubmitHandler<FieldValues>}>
         <div className="basic-info-form">
-          <FormFieldCreator form={formFields} />
+          {formFields && <FormFieldCreator form={formFields} />}
```

Comment:
Does this have any effect "formFields &&"? formFields by default contains empty fields thus it is always true.

Reply:
Fixed